encodings/fastlanes/src/for/compress.rs

```diff
@@ -103,11 +103,10 @@ fn decompress_primitive<T: NativePType + WrappingAdd + PrimInt>(
     shift: u8,
 ) -> Vec<T> {
     if shift > 0 {
-        let shifted_reference = reference << shift as usize;
         values
             .iter()
             .map(|&v| v << shift as usize)
-            .map(|v| v.wrapping_add(&shifted_reference))
+            .map(|v| v.wrapping_add(&reference))
```

Comment:
Looking at compress (see screenshot), the symmetric operation would be to keep current code but reverse the order. I think this is preferable to avoid any subtle bugs with `<<` truncation if we addition/subtraction end up wrapping (though that should be impossible? I think?)

<img width="301" alt="Screenshot 2024-08-14 at 21 41 42" src="https://github.com/user-attachments/assets/cb031ae7-db77-4ed9-9605-5c29bc6e58f5">


i.e., `.map(|v| v.wrapping_add(&shifted_reference)).map(|&v| v << shift as usize)`

Reply:
eh, this is fine, there's no truncation